### PR TITLE
[QBS] changes added to latest src main tree

### PR DESCRIPTION
**Pull request by quadrant2005**
1mb and 10mb addition to steps
more band definitions
and adding of callsign to menu system

This is my last attempt to get things intergrated cleanly into the main source tree via my fork, without messing up by including eclipse files etc. fingers crossed this time i get it right, as i do not want to anoy the main team members hopefully ive worked it out now,  regards Richard 2E0DVK

### CODE REVIEW

**Comment by db4ple** [comment]
Hi,
that is look much, much better now. In order to not get bothered by .metadata stuff, it should be added to the .gitignore file. I just made a pull request to include this in the main repository.

Now that I can see your commit a few comments come to my mind:

 * I don't understand the concept of the [QSB].txt file. Do you think, every commiter should maintain such a change log, that would make it very hard to follow what is going since one would have to mix all the files in the right order. BTW, git log does exactly this with no extra effort. If we go for a maintained changelog (shared by all), that is a different story, I can see a value in that for those "external" viewers not wanting to wade through the hundreds of git log entries but still interested in seeing the major changes. 

 * I personally dislike tagging almost every line with an identifier who added this line.  There is next to no value in this for code comprehension, more on the contrary. Comments should provide additional information or ease understanding the code, give rules for changing related code, etc. Everything else is unwanted noise. If I want to know who made a change, I ask git to tell me that (git blame). While also git blame alone is not sufficient, I can use this to get the whole history of an area of interest if necessary. 
Github has a fairly usable diff viewer, where it is easy to figure out what a pull request contains. And it is clear at this point who did the pull request and github remembers that (on the levewl of commits, not pull requests)

 * Please do not include "inactive" code unless there is a good reason for (e.g. if you want to share that code with others for experimenting. You can use "git stash" to handle this and for instance git-gui  to handle make commits where not a whole file is to be commited but only some changes (Eclipse cannot do this cool trick AFAIK). 

Rationale for all my comments given above: Do not create more "work" for other contributors with added value. 

Please don't see these comments as a way to tell you that you're not being welcome to contribute. It is the opposite. Part of this is to understand the rules of the game. Feel free to disagree and discuss.

**Comment by db4ple** [comment]
Just forgot one more: the title commit/pull request "changes added to latest src main tree" is not very helpful (well, it is expected that your changes are made on the latest source tree). The full commit message itself is good, btw.  Something like "Band Info additions, minor documentation updates" would give an good idea if one wants to dig deeper or just skip of this.
Before a pull request has been merged, you can change both the pull request information (using github web page) AND also change the commits in the pull request ( via `git amend`, https://www.atlassian.com/git/tutorials/rewriting-history and then `git push -f` ). You may want to try this.

**Comment by df8oe** [comment]
Hi,
agree to db4ple's statements. Additionally one of our contribution rules is that a pull request must give a benefit for a big group of users. I do not see a benefit like that here. And please do not add yourself to the "Hall of Fame" - that is a work which is done by the maintainer of the main repository ;)

So I will not merge this. But you can play around with this pull request to become familar with the working steps and the behaviour of this. I do not close - I let it rest here for playing around.

File: RemoteSystemsTempFiles/.project
Comment (db4ple) [inline RemoteSystemsTempFiles/.project]: I have no idea, what this added Eclipse nature means for others, please get rid of that or we have to be sure that this does not  create issues for other Eclipse users. Maybe someone else has an idea what this is?

File: mchf-eclipse/drivers/ui/menu/ui_menu.c
Comment (db4ple) [inline mchf-eclipse/drivers/ui/menu/ui_menu.c]: If you would use ```snprintf(out,32," 0x%08X%08X%08X\n", STM32_UUID[0],STM32_UUID[1],STM32_UUID[2]);```
you don't need 3 menu entries to do the job and we shoud have enough characters (about 33 in total AFAIR) . And in any case please move this menu item down to the end of the list. The UUID is a very infrequently required information, as it does not bear any technical relevance except ownership maybe as it is just that, a unique 96 bit number different for each STM32 MCU with al least based on my knowledge no additional relevant meaning for us.

File: mchf-eclipse/drivers/ui/ui_driver.c
Comment (db4ple) [inline mchf-eclipse/drivers/ui/ui_driver.c:1585]: We will not accept this, the UHSDR firmware is not a UK firmware so please do not put UK specific bandplan information in the official menu.

**Review by db4ple — CHANGES_REQUESTED** [review]
Hi,
please look at my comments. As a whole this pull request will not go through since it also includes the "old" content which is not fit for merging due to its too specialized nature and other issue already discussed. 
The CPU UUID commit alone would be (with changes, see below) acceptable. 

There are a number of ways how you can get rid of existing commits. One is to tell git to create a commit which "reverts" a specific commit or set of commits (https://stackoverflow.com/questions/1463340/revert-multiple-git-commits);
another one which is kind of feasible in your case is to reset your active-devel to the commit before the first of your commits you want to get rid of (git reset --hard HEAD~<number of commits to loose>) but this is dangerous unless you really know how git works as it if done the wrong way may create apparent loss of work (there are ways to fix this too, as git not really forgets things immediately)

File: mchf-eclipse/drivers/ui/ui_driver.c
Comment (quadrant2005) [reply mchf-eclipse/drivers/ui/ui_driver.c:1585]: Hi Guys, been really unwell this week, just starting to feel abit better today. Im going to have to tackle the commit 

suituation when im totally well. Yes i very much understand what you guys have said about what i submitted, I just need to 

learn how to only submit specific things to you guys. I think ive done well with the CPU UUID code as this acts as a unique 

serial number for the radio as its read only burned into the CPU at the factory. Ive now worked out how to keep my code 

inline with the main head fork, what i do is a pull request on my on fork when i see that there has been changes to the main 

head, i submit the changes in the head fork to my fork then accept them. So far so good :) if you guys would like the CPU 

UUID code ive pasted the two submits here the original then the one with the edited removed obsolete bits...  Thank you for 

being patient with me. I find it hard to take everything in with my health condition. But would very much like to contribute 

in some way.  I have a question would you guy be interested if i also added Elecraft K3 Cat control that is switchable in the menu system like the Yaesu Cat control code is, It would even be possible to have both enabled at the same time if wished. The reason for saying this is because there are some programs that are K3 compatible but not Yaesu compatible, There is a version of PowerSDR that works with the K3. Please find below the link to the two submits for the CPU UUID code
regards Richard 2E0DVK, ps i always tag my changes because it makes it very fast to find them in the code base as i just search [QBS] its the way me and a team worked back some 18 years or so ago i used to write mods for the Quake 2 computer game. I also wrote a full 2d/3d gaming engine back then, 

CPU UUID (Unique serial number)
https://github.com/quadrant2005/UHSDR/commit/372ae301d96c8f03f62586933964fc736e1d79e1
https://github.com/quadrant2005/UHSDR/commit/246603dab35bf5b7aade1cc57ad5dd8d490b49d9

**Comment by quadrant2005** [comment]
Hi there db4ple, not sure why that showed here as ive not submitted it to the head fork, Only made the possible bugfix in my own fork. But did submit details above for the UUID Code submit...

CPU UUID (Unique serial number)
quadrant2005/UHSDR@372ae30
quadrant2005/UHSDR@246603d
Currently trying to get my head arround KiCad as i wish to layout a UI board for a base station radio, using two STM32F427VIT6 Arm Processors and a esp8266mod having the  esp8266 with its analogue inputs will allow me to hang lots of buttons from one io pin, just use different resistor values so that the voltage level returned from the tactswitch will return a different value, the Esp8266 will also add wifi support for retrieving Band condition data the K index etc of what bands should be open, This will also allow for optional configuration from the web browser. Ive also a STM32_F4AVE Dev board that i would like to get an old version of the firmware up and running on, To do some experiments to make a stripped down version as a frontend interface for SSB6.1 kit radios etc, regards Richard. At some point i will ask my friend Andy if he can swop the 512k nand flash cpu that is on the STM32_F4AVE Dev Board for a 2mb nand STM32F427VIT6 for me which would make better sense.. cheers guys. At least ive worked out how to submit changes to the main head fork to my code base, Just wish i could learn to do the same with the tools that im using, to be able to submit specific updates to you guys for example the UUID code, The tools that im using are under windows 7 64bit  i use GitHub Desktop and Eclipse Neon.3 with me being unwell i need to use programs with a GUI, commandline things will not work for me my memory is just far to bad and ends up in me getting frustrated and upset when i try. Because in the past ive always been used to being able to multitask and work on lots of different things, But due to my health i find it hard to hold on two one thing at a time these days which i find upsetting..   Thanks Guys p.s is there a proper location where all the devs can communicate with each other that im unaware of or have forgotten.

**Comment by db4ple** [comment]
Hi,
all your commits are becoming part of the original pull request #1448 the moment you push them to your GitHub repository. This is a "GitHub" thing to make it easy for all of us to easily amend a pull request. 
Close the pull request and this magic will be gone.

**Comment by quadrant2005** [comment]
db4ple  hehe ok :) before i close this please where is it appropriate to communicate with the dev team. ;-)

**Comment by quadrant2005** [comment]
Totally a long shot but did anyone download my code base, lost everything the 2.9.10 pull got me in a mess.  I recovered my fork, but applied the bugfix that db4ple posted this fixed the code base, then when 2.9.11 of the main fork went up, i could no longer merge, i removed the changes in my file but still would not let me merge.  I moved my work into another folder, in the end deleted my fork and reforked, then sync to my computer then ment to copy my work into the UHSDR folder so git desktop would see my changes and intergrate them with my fork but i messed up i copied the folder the wrong way and over wrote my work, I had rar the folder before doing this but once again it was your code i backed up not mine. I hate being sick it really sucks :( lost everything.. I can not believe i did it,  Ive managed to pull changes upto the 14th of feb from this pull request, but lost 14 days work, i got two devices working on the HOST port lastnight it requires some changes in one of the files because is it cubemx or something doesnt handle the second host it put it all in one structure and gets over written, i had working keyboard and mouse lastnight so so totally gutted...  i made a massive code push yesterday its my fault but feel so upset....   I did keep backups of all my work but on a usb pen drive which got damaged the other day because i dropped my laptop and it was plugged in the side, Just having a bad run of things at the moment here...

**Pull request closed without merging**
